projects/buttons_and_adc/src/main.c

```diff
@@ -0,0 +1,61 @@
+// Created by Sudhish M on 2020-10-04.
+// buttons_and_adc
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static GpioAddress adc_pin_address{ .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress button_address{ .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings button_settings = {
+  .direction = GPIO_DIR_IN,
+  .resistor = PIO_RES_PULLDOWN,
+  // Reason for using pulldown: when no signal is received
+  // from the button (unpressed), the value sent to the pin is 0
+  .alt_function = GPIO_ALTFN_NONE
+};
+
+static GpioSettings adc_pin_settings = {
+  .direction GPIO_DIR_IN, .resistor = GPIO_RES_NONE, .alt_function = GPIO_ALTFN_ANALOG
+};
```

Comment:
If everything being on one line was the result of a `make format`, you can add `//` at the end of each line to force it not to pack all the lines; otherwise, I'd prefer each field to be on a different line, with a comma on the end of the last line (GCC supports this). Also, of course, don't forget the `s_` on `adc_pin_settings`.